src/main/js/helpers/axis.js

```diff
@@ -175,17 +175,33 @@ const createAxisReferenceLine = (axis, scale, config, canvasSVG) => {
  * @returns {object} - Scaled axes object
  */
 const getAxesScale = (axis, scale, config) => {
+    let tickFormatToTrimTrailingZeros;
+
+    if (
+        config.suppressTrailingZeros &&
+        config.axis.x.type === AXIS_TYPE.DEFAULT &&
+        utils.isUndefined(config.axis.x.ticks.format)
```

Comment:
Can you add a comment on why we are doing this way?

You have added below scenario here for x-axis
- If the consumer

    1. `suppressTrailingZeros` prop is provided
    2. when axis type is default
    3. x-axis ticks format is undefined

Are we going to cover the below scenarios as well here?

What would happen if 

- If the consumer 

    1. `suppressTrailingZeros` prop is provided
    2. when axis type is default
    3. x-axis ticks format is **defined**

What would happen if 

- If the consumer

    1. `suppressTrailingZeros` prop is **not** provided
    2. when axis type is default
    3. x-axis ticks format is **defined**

Reply:
Added comment [here](https://github.com/JananiGunasekaran/carbon-graphs/commit/4d775051c01587056c48fd5f9ad5d05699b247e3).

All the scenarios mentioned above are being covered in the code:

1. If the consumer 

- suppressTrailingZeros prop is **provided**
- when axis type is default
- x-axis ticks format is **defined**

------- The if condition will not get satisfied and it will execute the next statement after the loop [here](https://github.com/JananiGunasekaran/carbon-graphs/commit/19249fbfd155a12c377f2b9c1f59042777167551#diff-c116b52d2491f4d3b18768a42e2a7a23R191). This statement will use the x-axis tick format provided by the consumer to format the tick values of the axis. As mentioned in the [ticks.md](https://github.com/JananiGunasekaran/carbon-graphs/commit/19249fbfd155a12c377f2b9c1f59042777167551#diff-8a20b2ec7dd8fdd32c7802f2fcdee2b6R29) docs, the consumer must specify the '~' character if they want to suppress the trailing zeros when they specify the tick format.

2. If the consumer

- suppressTrailingZeros prop is **not provided**
- when axis type is default
- x-axis ticks format is **defined**

------- The if condition will not get satisfied and hence it will execute the next statement which will use the x-axis tick format provided by the consumer to format the tick values of the axis. If the consumer is going to specify the tick format, they have the option to decide whether they want to suppress the trailing zeros or not, by specifying the '~' character in their tick format.

3. There is also another scenario:
If the consumer

- suppressTrailingZeros prop is **not provided**
- when axis type is default
- x-axis ticks format is **undefined**

-------- The if condition will not get satisfied and hence it execute the next statement which will just use the default d3 tick format to format the tick values without suppressing the trailing zeros since both suppressTrailingZeros is not provided and x-axis ticks format is undefined.